Spec/Auth.swift

```diff
@@ -3725,47 +3725,37 @@ class Auth : QuickSpec {
                         connectedStateCount += 1
                     }
 
-                    var tokenDetailsFirst: ARTTokenDetails?
                     var tokenDetailsLast: ARTTokenDetails?
                     waitUntil(timeout: testTimeout) { done in
                         let partialDone = AblyTests.splitDone(2, done: done)
-                        realtime.auth.authorize { tokenDetails, error in
-                            if let error = error, (error as NSError).code != URLError.cancelled.rawValue {
-                                fail(error.localizedDescription); partialDone(); return
-                            }
-                            expect(tokenDetails).toNot(beNil())
-                            if tokenDetailsFirst == nil {
-                                tokenDetailsFirst = tokenDetails
-                            }
-                            else {
-                                tokenDetailsLast = tokenDetails
-                            }
-                            partialDone()
-                        }
-                        realtime.auth.authorize { tokenDetails, error in
-                            if let error = error, (error as NSError).code != URLError.cancelled.rawValue {
-                                fail(error.localizedDescription); partialDone(); return
-                            }
-                            expect(tokenDetails).toNot(beNil())
-                            if tokenDetailsFirst == nil {
-                                tokenDetailsFirst = tokenDetails
+                        let callback: (ARTTokenDetails?, Error?) -> Void = { tokenDetails, error in
+                            if let error = error {
+                                if (error as NSError).code == URLError.cancelled.rawValue {
+                                    expect(tokenDetails).to(beNil())
+                                }
+                                else {
+                                    fail(error.localizedDescription); partialDone(); return
+                                }
                             }
                             else {
+                                expect(tokenDetails).toNot(beNil())
                                 tokenDetailsLast = tokenDetails
                             }
                             partialDone()
                         }
+                        // One of them will be canceled by the connection:
+                        realtime.auth.authorize(callback)
+                        realtime.auth.authorize(callback)
                     }
 
-                    expect(tokenDetailsFirst?.token).toNot(equal(tokenDetailsLast?.token))
                     expect(realtime.auth.tokenDetails).to(beIdenticalTo(tokenDetailsLast))
                     expect(realtime.auth.tokenDetails?.token).to(equal(tokenDetailsLast?.token))
 
```

Comment:
Originally `tokenDetailsLast` made sense only in a duet with tokenDetailsFirst, meaning "The first call was cancelled by the second, and the result contained in `realtime.auth.tokenDetails` is from the second call". Without the `tokenDetailsFirst` this check does nothing, except ensuring `realtime.auth.tokenDetails` is the same thing that was thrown in the callback param, which is not the purpose of this check as I can guess.

Reply:
Ok, I've added one flag to ensure that the previous authorization callback was cancelled by next. The rest of these looks good to me.

Reply:
changed here 7e23fe5501b4956a5f33ff070f5311a41d912aa2

Reply:
Hmm.. that doesn't check whether the second call cancels the first one. This callback potentially can be called by the first call and set `didCancelAuthorization` to `true`. And this is what originally this test was trying to ensure library doesn't do.

Reply:
As I understand this, in this test, one of these two requests should be cancelled, but it's no matter which one